builtin/game/item.lua

```diff
@@ -227,6 +239,27 @@ function core.get_node_drops(node, toolname)
 				end
 			end
 		end
+		if item.tool_groups ~= nil and toolname then
+			local tooldef = core.registered_items[toolname]
+			if tooldef ~= nil and type(tooldef.groups) == "table" then
+				if type(item.tool_groups) == "string" then
```

Comment:
One more thing: you can fold all conditions into a single if to avoid one indentation level
```lua
			if not tooldef or not tooldef.groups then
				good_tool = false
			elseif type(item.tool_groups) == "string" then
			-- [...]
```

Reply:
How attached to this suggestion are you?

To me it looks like increasing cognitive load: making readers consider a negative situation which doesn't matter, mentally inverting later on for the condition that does matter, adding an assignment that isn't necessary, and possibly causing devs to question assumptions upon reaching an unnecessary assignment.

However, I'm not maintaining `item.lua` so my line length preferences don't matter, if you're leaning toward fewer tabs then I'll get this change in.

Reply:
> How attached to this suggestion are you?

Not very, if you don't agree don't make the change.

> adding an assignment that isn't necessary

By the way this is just so luacheck doesn't complain about an empty `if` body.